bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/XPathFlex.java

```diff
@@ -853,6 +854,10 @@ private void zzDoEOF() throws java.io.IOException {
       zzEOFDone = true;
       yyclose();
     }
+
+	if (commentLevel > 0) {
+		throw new XPathParserException("Unterminated comment", null);
+	}
```

Comment:
This is generated code, the modification should probably be placed in the grammar file...under ```grammars/xpath.lex```

````flex
  %eof{
        ...
   %eof}
````